### DISCUSSION TITLE
rust-analyzer is not installed for the toolchain

### DISCUSSION THREAD

**Discussion by darkseid-is** [mrcjkb/rustaceanvim#256] [Q&A] [answered]
### Neovim version (nvim -v)

v0.9.5

### Operating system/version

Debian

### Output of :checkhealth rustaceanvim

```console
==============================================================================
rustaceanvim: require("rustaceanvim.health").check()

Checking for Lua dependencies ~
- OK [mfussenegger/nvim-dap](https://github.com/mfussenegger/nvim-dap) installed.

Checking external dependencies ~
- OK rust-analyzer: found
- OK Cargo: found cargo 1.76.0 (c84b36747 2024-01-18)
- OK rustc: found rustc 1.76.0 (07dca489a 2024-02-04)

Checking config ~
- OK No errors found in config.

Checking for conflicting plugins ~
- OK No conflicting plugins detected.

Checking for tree-sitter parser ~
- OK tree-sitter parser for Rust detected.

error: 'rust-analyzer' is not installed for the toolchain 'stable-x86_64-unknown-linux-gnu'
```

### How to reproduce the issue

```console
Install rust
Download rust-analyzer binary to path
```

### Expected behaviour

Load rust-analyzer binary from path.

### Actual behaviour

error: 'rust-analyzer' is not installed for the toolchain 'stable-x86_64-unknown-linux-gnu'

### The minimal config used to reproduce this issue.

```Lua
Default config
```

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

This is not a rustaceanvim bug, but an issue with how you are installing rust-analyzer.

Please refer to the [rust-analyzer installation manual](https://rust-analyzer.github.io/manual.html#installation):

As of today, you need to run:

```sh
rustup component add rust-src
rustup component add rust-analyzer
```

The `rust-analyzer` that rustaceanvim detected in the health check is just the wrapper provided by rustup, which launches the real executable for your system.

Closing and converting up a discussion.

**Reply by darkseid-is** [reply]
Thanks for a great plugin and prompt response. Looking out for bin support  support in future release. Thank You.
